CMakeLists.txt

```diff
@@ -143,7 +137,7 @@ if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
 
     include(CMakePackageConfigHelpers)
     write_basic_package_version_File("otf2xxConfigVersion.cmake"
-        VERSION ${otf2xx_VERSION}
+        VERSION ${PROJECT_VERSION}
```

Comment:
Is there a compelling reason to use ${PROJECT_...} variables in general? It's neither more portable, because the same statement contains `otf2xx` anyways nor is it shorter.

Reply:
There is probably a good guideline for that but what I can think of:
- Shorter as no `set(otf2xx_VERSION 1.0)` required, instead added to `project` command
- Standard names, so less confusion (OTF2XX_VERSION or otf2xx_VERSION?) and better portability (e.g. in `write_basic_package_version_File` you just need to change the name once)
- more version variables for free: `PROJECT_VERSION, <PROJECT-NAME>_VERSION, PROJECT_VERSION_MAJOR, `<PROJECT-NAME>_VERSION_MAJOR,...`

So just minor reasons, but IMO the `PROJECT_` variables are better for reusing CMake code. Note that you can e.g. use `write_basic_package_version_File("${PROJECT_NAME}ConfigVersion.cmake" ...`

Final note: Adding the version to `project(otf2xx` would also allow using `${otf2xx_VERSION}`.

Reply:
> Final note: Adding the version to project(otf2xx would also allow using ${otf2xx_VERSION}.

That's why I'm asking...

Reply:
Ok didn't got that from the question.
I prefer the `PROJECT_*` variables (including `PROJECT_NAME`, `PROJECT_BINARY/SOURCE_DIR`...), as it makes the code more generic. One could put the packaging code into an own function and reuse it. I'm surprised `CMakePackageConfigHelpers` does not do that.